File: PE22/PE22.cpp
// PE22.cpp : This file contains the 'main' function. Program execution begins and ends there.
//

#include "pch.h"
#include <iostream>
#include <string>
#include <fstream>
#include <vector>
#include <algorithm>
#include <sstream>

using namespace std;

vector<string> split(const string &s, char delim) {
	stringstream ss(s);
	string item;
	vector<string> elems;
	while (getline(ss, item, delim)) {
		elems.push_back(item);
	}
	return elems;
}

int strval(string s) {
	int val = 0;
	for (char c : s) val += c - 'A' + 1;
	return val;
}
int main() {
	ifstream f ("names.txt");
	string line; getline(f, line);
	vector<string> fcomma = split(line, ',');
	f.close();

	vector<string> mylist;
	for (string s : fcomma) {
		mylist.push_back(s.substr(1, s.length() - 2));
	}
	sort(mylist.begin(), mylist.end());

	int mysum = 0;
	int i = 1;
	for (string s : mylist) {
		mysum += i * strval(s);
		i++;
	}

	cout << mysum << endl;
}

File: PE22/PE22.py
def strval(s):
    val = 0
    for c in s:
        val += ord(c) - ord('A') + 1
    return val

f = open("names.txt", 'r')

fcomma = f.readline().split(",")
f.close()

mylist = []
for s in fcomma:
    mylist.append(s[1:len(s)-1])

mylist = sorted(mylist)

mysum = 0
i = 1
for s in mylist:
    mysum += i * strval(s)
    i += 1

print(mysum)
